### PR TITLE
ARROW-7505: [Java] Remove Netty dependency for ArrowBuf

### PR DESCRIPTION
**Pull request by liyafan82**
This is part of the first step of issue ARROW-4526.
In this step, we remove netty dependency for ArrowBuf, BufferAllocator and ReferenceManager.

In this issue, we remove the dependency for ArrowBuf.
The task for BufferAllocator and ReferenceManager will not start until ARROW-7329 is finished.

### CODE REVIEW

**Comment by liyafan82** [comment]
@jacques-n Would you please take a look?

**Comment by emkornfield** [comment]
I checked in the int64 address space change so this PR and probably a few other might need to be rebased.

**Comment by liyafan82** [comment]
> I checked in the int64 address space change so this PR and probably a few other might need to be rebased.

Sure. Thanks for your kind reminder.

**Comment by emkornfield** [comment]
@jacques-n or @rymurr do you have bandwidth to at least do a first pass review?

File: java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   */
+  public static final long BYTE_ARRAY_BASE_OFFSET;
+
+  /**
+   * The offset of the address field with the {@link java.nio.ByteBuffer} object.
+   */
+  static final long BYTE_BUFFER_ADDRESS_OFFSET;
+
+  static {
+    // get the unsafe object
```
Comment (siddharthteotia) [inline java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java]: Should we try to mimic exactly how it is done in PlatformDependent0?

Comment (siddharthteotia) [reply java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java]: I mean do we need all the checks that done there?

Comment (liyafan82) [reply java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java]: Thanks a lot for your comments. I will try to mimic the logic in PlatformDependent0.

Comment (liyafan82) [reply java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java]: @siddharthteotia I have revised the code to mimic PlatformDependent implementation. Would you please take another look?

**Comment by siddharthteotia** [comment]
@liyafan82 , the changes look good to me. Looks like there is a build failure

**Review by siddharthteotia — APPROVED** [review]

**Comment by liyafan82** [comment]
> @liyafan82 , the changes look good to me. Looks like there is a build failure

@siddharthteotia Thanks a lot for your effort. It seems all my PRs are experiencing this failure. I will do a rebase when the problem is resolved.

**Comment by emkornfield** [comment]
@siddharthteotia can this be merged, i think the integration test is unrelated?

**Comment by siddharthteotia** [comment]
Thanks, @liyafan82 . Merging this as the integration test failure seems to be unrelated. @liyafan82 , it would be great if you can keep an eye on the build and.make sure the build on master is clean after this PR is merged (just in case the integration test failure happens to be caused by this change)

**Comment by liyafan82** [comment]
> Thanks, @liyafan82 . Merging this as the integration test failure seems to be unrelated. @liyafan82 , it would be great if you can keep an eye on the build and make sure the build on master is clean after this PR is merged (just in case the integration test failure happens to be caused by this change)

@siddharthteotia Thanks a lot for your effort. I will keep an eye on the master branch.

**Pull request merged**